gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt

```diff
@@ -143,37 +143,37 @@ internal class AvatarPickerViewModel(
                     failedUploadDialog = null,
                 )
             }
-            when (val result = avatarRepository.uploadAvatar(email, uri)) {
+            val selectAvatar = _uiState.value.emailAvatars?.selectedAvatarId == null
+            when (val result = avatarRepository.uploadAvatar(email, uri, selectAvatar)) {
```

Comment:
> "description": "Determines if the uploaded image should be set as the avatar for the email. If not passed, the image is only selected as the email's avatar if no previous avatar has been set. Accepts '1'/'true' to always set the avatar or '0'/'false' to never set the avatar.",

❓ Based on that, I wonder If we should not pass any value to "selectAvatar" and let the backend handle the default action. With that code, we are forcing to select the avatar but I think that's unnecessary as, if there is no avatar selected, the backend will select it by default.

Reply:
Makes sense! I realized I should have make the `hash` and `selectAvatar` nullable in the `AvatarService` so I've updated that as well.